homeassistant/components/fireservicerota/switch.py

```diff
@@ -0,0 +1,137 @@
+        self._state_attributes = {}
+
+    @property
+    def name(self) -> str:
+        """Return the name of the switch."""
+        return "Incident Response"
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        return "mdi:forum"
+
+    @property
+    def is_on(self) -> bool:
+        """Get the assumed state of the switch."""
+        return self._state
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this switch."""
+        return self._unique_id
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def device_state_attributes(self) -> object:
+        """Return available attributes for switch."""
+        attr = {}
+        if not self._state_attributes:
+            return attr
+
+        data = self._state_attributes
+        attr = {
+            key: data[key]
+            for key in (
+                "user_name",
+                "assigned_skill_ids",
+                "responded_at",
+                "start_time",
+                "status",
+                "reported_status",
+                "arrived_at_station",
+                "available_at_incident_creation",
+                "active_duty_function_ids",
+            )
+            if key in data
+        }
+
+        _LOGGER.debug("Set attributes of entity 'Response Switch' to '%s'", attr)
+        return attr
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Send Acknowlegde response status."""
+        await self.async_set_response(True)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Send Reject response status."""
+        await self.async_set_response(False)
+
+    async def async_set_response(self, value) -> None:
+        """Send response status."""
+        if not self._client.on_duty:
+            _LOGGER.debug(
+                "Cannot send incident response when not on duty",
+            )
+            return
+
+        await self._client.async_set_response(value)
+        self.client_update()
+
+    async def async_added_to_hass(self) -> None:
+        """Register update callback."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{FIRESERVICEROTA_DOMAIN}_{self._entry_id}_update",
+                self.client_update,
+            )
+        )
+
+    @callback
+    def client_update(self) -> None:
+        """Handle updated incident data from the client."""
+        self.async_schedule_update_ha_state(True)
+
+    async def async_update(self) -> bool:
+        """Update FireServiceRota response data."""
+        data = await self._client.async_response_update()
+
+        if not data or "status" not in data:
+            return
+
+        if data["status"] == "acknowledged":
+            self._state = True
+        else:
+            self._state = False
+
+        del data["user_photo"]
```

Comment:
Why delete something when you're already filtering keys in device_state_attributes?

Reply:
Good one, it's old code before I weeded out the keys, Frenck made me limit the amount earlier ;-)
Will fix.

Reply:
Fixed